extensions/wikia/Search/classes/TopWikiArticles.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Wikia\Search;
+
+class TopWikiArticles {
+	/**
+	 * Used for changing mem cache key for top articles snippet (after changed will end up in purging cache)
+	 */
+	const TOP_ARTICLES_CACHE = 1;
+
+	/**
+	 * 5 days, one business week
+	 */
+	const CACHE_TTL = 86400 * 5;
+
+	/**
+	 * We want to return articles with thumbnails, but not all articles
+	 * returned from ArticlesApiController:getTop() have them, that is why we ask for
+	 * more articles than we will return
+	 */
+	const GET_NUMBER_OF_TOP_ARTICLES_FROM_DB = 12;
+	const TOP_ARTICLES_RESULT_LIMIT = 5;
+
+	static public function getArticlesWithCache( $cityId, $isGridLayoutEnabled ) {
+		$cacheKey = wfMemcKey(
+			__CLASS__,
+			'WikiaSearch',
+			'topWikiArticles',
+			$cityId,
+			static::TOP_ARTICLES_CACHE,
+			$isGridLayoutEnabled
+		);
+
+		return \WikiaDataAccess::cache(
+			$cacheKey,
+			86400 * 5, // 5 days, one business week
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)